velox/exec/OutputBuffer.h

```diff
@@ -78,6 +81,9 @@ class ArbitraryBuffer {
   /// there are sufficient buffered pages.
   std::vector<std::shared_ptr<SerializedPage>> getPages(uint64_t maxBytes);
 
+  /// Append the available page sizes to `out'.
+  void getAvailablePageSizes(std::vector<int64_t>& out) const;
```

Comment:
Any particular reason not to return std::vector?

Reply:
So we don't need to create a new vector every time and just appending the existing one already created for DestinationBuffer